packages/cli/src/commands/hydrogen/upgrade.ts

```diff
@@ -0,0 +1,2931 @@
+
+  const upgradingReleases = releases.filter((release) => {
+    const isUpgrade =
+      semver.gt(release.version, currentPinnedVersion) &&
+      semver.lte(release.version, selectedRelease.version);
+
+    if (!isUpgrade) return false;
+
+    return true;
+  });
+
+  return upgradingReleases.reduce(
+    (acc, release) => {
+      acc.features = [...acc.features, ...release.features];
+      acc.fixes = [...acc.fixes, ...release.fixes];
+      return acc;
+    },
+    {features: [], fixes: []} as CummulativeRelease,
+  );
+}
+
+/**
+ * Displays a confirmation prompt to the user with a list of features and fixes
+ * included in the upgrade versions range. The user can also return to the
+ * version selection prompt if they want to choose a different version.
+ **/
+export async function displayConfirmation({
+  appPath,
+  cumulativeRelease,
+  selectedRelease,
+  targetVersion,
+  dryRun,
+}: {
+  appPath: string;
+  cumulativeRelease: CummulativeRelease;
+  selectedRelease: Release;
+  targetVersion?: string;
+  dryRun: boolean;
+}) {
+  const {features, fixes} = cumulativeRelease;
+  if (features.length || fixes.length) {
+    renderInfo({
+      headline: `Included in this upgrade:`,
+      //@ts-ignore we know that filter(Boolean) will always return an array
+      customSections: [
+        features.length && {
+          title: 'Features',
+          body: [
+            {
+              list: {
+                items: features.map((item) => item.title),
+              },
+            },
+          ],
+        },
+        fixes.length && {
+          title: 'Fixes',
+          body: [
+            {
+              list: {
+                items: fixes.map((item) => item.title),
+              },
+            },
+          ],
+        },
+      ].filter(Boolean),
+    });
+  }
+
+  const confirmedUpgrade = await renderConfirmationPrompt({
+    message: `Are you sure you want to upgrade to ${selectedRelease.version}?`,
+    cancellationMessage: `No, choose another version`,
+    defaultValue: true,
+  });
+
+  if (!confirmedUpgrade) {
+    // Go back to the version selection prompt
+    return await runUpgrade({
+      appPath,
+      dryRun,
+      version: targetVersion,
+    });
+  }
+
+  return true;
+}
+
+function isRemixDependency([name]: [string, string]) {
+  if (name.includes('@remix-run')) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Builds the arguments for the `npm|yarn|pnpm install` command
+ */
+export function buildUpgradeCommandArgs({
+  selectedRelease,
+  currentDependencies,
+}: {
+  selectedRelease: Release;
+  currentDependencies: Dependencies;
+}) {
+  const args: string[] = [];
+
+  // upgrade dependencies
+  for (const [name, version] of Object.entries(selectedRelease.dependencies)) {
+    const isRemixPackage = isRemixDependency([name, version]);
+    const dependencyExists = currentDependencies[name];
+    if (isRemixPackage || !dependencyExists) continue;
```

Comment:
I guess we skip Remix dependencies because it's now a peer dep? In that case, shouldn't we just skip including Remix in the changelog directly?

However, I think it might still be useful to include it and at least warn about mismatches. For example, if we change our minimum peer dep or Remix to XYZ and the user has a lower version, our new Hydrogen version might not work.
I think this hasn't happened yet so maybe it's OK to skip for now and implement when needed. Thoughts?